pyccel/codegen/printing/fcode.py

```diff
@@ -1787,8 +1787,8 @@ def _print_AugAssign(self, expr):
 
     def _print_PythonRange(self, expr):
         start = self._print(expr.start)
-        stop  = self._print(expr.stop) + '-' + self._print(LiteralInteger(1))
         step  = self._print(expr.step)
+        stop = self._print(expr.stop) + '- ( (merge(1_C_INT64_T, -1_C_INT64_T, ' + self._print(expr.step) + ' > 0_C_INT64_T)) *' + self._print(LiteralInteger(1)) + ')'
```

Comment:
you can use``` self._print(LiteralInteger(1)``` instead of ```1_C_INT64_T```
                   ``` self._print(LiteralInteger(-1)``` instead of ```-1_C_INT64_T```
and let the printed precision  depend on the system used .

like in my previous comment you should try to use PyccelNods , they make code look clearer and readable.

you can use formatted code to clean the generated code example ```hello {1}'.format('world')```